recipes/gstreamer-full/all/conanfile.py

```diff
@@ -0,0 +1,864 @@
+
+        # conan center packages / system
+        "with_egl": [True, False],
+        "with_wayland": [True, False],
+        "with_xorg": [True, False],
+    }
+    options.update({f'gst_base_{_name}': [True, False] for _name in GST_BASE_MESON_OPTIONS})
+    options.update({f'gst_base_{_name}': [True, False] for _name in GST_BASE_MESON_OPTIONS_WITH_EXT_DEPS})
+    options.update({f'gst_base_{_name}': [True, False] for _name in GST_BASE_MESON_OPTIONS_GL})
+    options.update({f'gst_good_{_name}': [True, False] for _name in GST_GOOD_MESON_OPTIONS})
+    options.update({f'gst_bad_{_name}': [True, False] for _name in GST_BAD_MESON_OPTIONS})
+    options.update({f'gst_ugly_{_name}': [True, False] for _name in GST_UGLY_MESON_OPTIONS})
+    options.update({f'gst_rtsp_server_{_name}': [True, False] for _name in GST_RTSP_SERVER_MESON_OPTIONS})
+
+    default_options = {
+        "with_base": True,
+        "with_good": True,
+        "with_ugly": True,
+        "with_bad": True,
+        "with_libav": True,
+        "with_rtsp_server": True,
+
+        "with_orc": True,
+        "with_introspection": False, # 1.72 is not yet compatible with conan 2.0
+        "with_coretracers": False,
+
+        "with_tools": False,
+
+        "gst_base_audioresample_format": "auto",
+        "gst_base_gl_jpeg": "libjpeg",
+
+        "with_egl": True,
+        "with_wayland": True,
+        "with_xorg": True,
+    }
+    default_options.update({f'gst_base_{_name}': True for _name in GST_BASE_MESON_OPTIONS})
+    default_options.update({f'gst_base_{_name}': True for _name in GST_BASE_MESON_OPTIONS_WITH_EXT_DEPS})
+    default_options.update({f'gst_base_{_name}': True for _name in GST_BASE_MESON_OPTIONS_GL})
+    default_options.update({f'gst_good_{_name}': True for _name in GST_GOOD_MESON_OPTIONS})
+    default_options.update({f'gst_bad_{_name}': False for _name in GST_BAD_MESON_OPTIONS})
+    default_options.update({f'gst_ugly_{_name}': False for _name in GST_UGLY_MESON_OPTIONS})
+    default_options.update({f'gst_rtsp_server_{_name}': False for _name in GST_RTSP_SERVER_MESON_OPTIONS})
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            self.options.rm_safe('with_wayland')
+            self.options.rm_safe('with_alsa')
+            self.options.rm_safe('gst_base_x11')
+            self.options.rm_safe('gst_base_xvideo')
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            self.options.rm_safe('with_egl')
+            self.options.rm_safe('with_xorg')
+
+        if not self.options.get_safe("with_xorg"):
+            self.options.rm_safe('gst_base_x11')
+            self.options.rm_safe('gst_base_xshm')
+            self.options.rm_safe('gst_base_xi')
+            self.options.rm_safe('gst_base_ximage')
+            self.options.rm_safe('gst_base_xvimage')
+
+        if not self.options.with_base:
```

Comment:
This won't work, unfortunately. The user-provided values for options are not available in `config_options()` or even `configure()`. You will have to `del self.info.options.xyz' the options that have no effect in `package_id()` instead.

Reply:
I'm not sure I follow. [Doc](https://docs.conan.io/2/reference/conanfile/methods/configure.html) seems to do it in configure(), see second snippet where they remove fPIC. 

I have no issue with moving it to package_id(), just want to understand a bit better. Thanks

Reply:
@danimtb Maybe you or someone else from the team can clarify whether the `configure()` step can be used like that or not, given that you were wrestling with similar issues in the Arrow recipe. Does the standard `if self.options.shared: del self.options.fPIC` even work as intended?

Reply:
I'm re-reading the [doc](https://docs.conan.io/2/reference/conanfile/methods/configure.html) and I think it should be in configure() and not in config_option() (that was a mistake as you pointed out, only settings is populated), and not in package_id(). I'll wait for confirmation on this. Thanks